### DISCUSSION TITLE
Hyp3 zip files to mintpy

### DISCUSSION THREAD

**Discussion by MocioVileda** [insarlab/MintPy#986] [Q&A]
Good morning everyone,
I'm new in this environment and I'm struggling to open ASF-Hyp3 products in Mintpy. 
I'm referring to this tutorial https://github.com/insarlab/MintPy-tutorial/blob/main/smallbaselineApp_hyp3.ipynb but I can't figure out how to prepare the input dataset having the .zip folders of my InSAR products.  The tutorial indicates a single "zip_file_name" ('Ridgecrest.zip') while I have more than one (one for each InSAR couple that Hyp3 processed).
What should I do? Unzip all the folders and gather all the products in a unique huge zip one?
And also, is it necessary to clip all the files on a common extent? Some other tutorials did it, but in the one I linked above it is not mentioned. 

Thanks a lot for your help!
M.V.

**Comment by asjohnston-asf** [comment]
The HyP3 team recommends the [Time series analysis with HyP3 and MintPy](https://nbviewer.org/github/ASFHyP3/hyp3-docs/blob/main/docs/tutorials/hyp3_insar_stack_for_ts_analysis.ipynb) notebook linked from the [Tutorials](https://hyp3-docs.asf.alaska.edu/tutorials/) page of our documentation. It covers requesting and preparing data from HyP3 in greater depth.

If you've already generated and downloaded your InSAR products, you can pick up with the last cell of Section 2, unzipping those products into a common folder, so you'll have a directory structure similar to:

```
data_dir/
  S1AA_20230129T033427_20230210T033426_VVR012_INT80_G_weF_534B/
    S1AA_20230129T033427_20230210T033426_VVR012_INT80_G_weF_534B_water_mask.tif
    S1AA_20230129T033427_20230210T033426_VVR012_INT80_G_weF_534B_corr.tif
    S1AA_20230129T033427_20230210T033426_VVR012_INT80_G_weF_534B_unw_phase.tif
    S1AA_20230129T033427_20230210T033426_VVR012_INT80_G_weF_534B_dem.tif
    S1AA_20230129T033427_20230210T033426_VVR012_INT80_G_weF_534B_lv_theta.tif
    S1AA_20230129T033427_20230210T033426_VVR012_INT80_G_weF_534B_lv_phi.tif
  S1AA_20230129T033452_20230210T033451_VVR012_INT80_G_weF_85A3/
    S1AA_20230129T033452_20230210T033451_VVR012_INT80_G_weF_85A3_water_mask.tif
    S1AA_20230129T033452_20230210T033451_VVR012_INT80_G_weF_85A3_corr.tif
    S1AA_20230129T033452_20230210T033451_VVR012_INT80_G_weF_85A3_unw_phase.tif
    S1AA_20230129T033452_20230210T033451_VVR012_INT80_G_weF_85A3_dem.tif
    S1AA_20230129T033452_20230210T033451_VVR012_INT80_G_weF_85A3_lv_theta.tif
    S1AA_20230129T033452_20230210T033451_VVR012_INT80_G_weF_85A3_lv_phi.tif
```

Yes, clipping the geotiffs to a common extent is required, as described in Section 3.1.

**Reply by MocioVileda** [reply]
Thanks @asjohnston-asf for your reply!
Since I was still having problems in managing the folder structure I decided to run the HyP3-MintPy notebook from the start as reported in this [tutorial](https://nbviewer.org/github/ASFHyP3/hyp3-docs/blob/main/docs/tutorials/hyp3_insar_stack_for_ts_analysis.ipynb) . However I still encounter some problems.
1) First: running the 3.2 section - Create the MintPy config file I get this: 
![image](https://user-images.githubusercontent.com/129189498/228500860-f4711653-c71f-4f5a-83d0-95644bd4049a.png)
Whats does the 688 stand for?

2) Then, in running section 3.3- run MintPy to do the time series analysis I get these lines of error:

C:\Users\xxx\AppData\Local\anaconda3\lib\site-packages\mintpy\objects\stack.py:917: RuntimeWarning: Mean of empty slice
  dmean[i] = np.nanmean(data)
C:\Users\xxx\AppData\Local\anaconda3\lib\site-packages\mintpy\utils\plot.py:532: RuntimeWarning: All-NaN slice encountered
  ax.bar(x_list, np.nanmax(coh_mat, axis=0), bar_width.days, label='Max {}'.format(p_dict['ds_name']))
C:\Users\xxx\AppData\Local\anaconda3\lib\site-packages\mintpy\utils\plot.py:533: RuntimeWarning: All-NaN slice encountered
  ax.bar(x_list, np.nanmin(coh_mat, axis=0), bar_width.days, label='Min {}'.format(p_dict['ds_name']))
Traceback (most recent call last):
  File "C:\Users\xxx\AppData\Local\anaconda3\Scripts\smallbaselineApp.py-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Users\xxx\AppData\Local\anaconda3\lib\site-packages\mintpy\cli\smallbaselineApp.py", line 208, in main
    run_smallbaselineApp(inps)
  File "C:\Users\xxx\AppData\Local\anaconda3\lib\site-packages\mintpy\smallbaselineApp.py", line 1117, in run_smallbaselineApp
    app.run(steps=inps.runSteps)
  File "C:\Users\xxx\AppData\Local\anaconda3\lib\site-packages\mintpy\smallbaselineApp.py", line 889, in run
    self.run_network_inversion(sname)
  File "C:\Users\xxx\AppData\Local\anaconda3\lib\site-packages\mintpy\smallbaselineApp.py", line 387, in run_network_inversion
    self.generate_temporal_coherence_mask()
  File "C:\Users\xxx\AppData\Local\anaconda3\lib\site-packages\mintpy\smallbaselineApp.py", line 449, in generate_temporal_coherence_mask
    raise RuntimeError(msg)
RuntimeError: Not enough reliable pixels (minimum of 100). Try the following:
1) Check the reference pixel and make sure it's not in areas with unwrapping errors
2) Check the network and make sure it's fully connected without subsets

Do you have any possible explanation for this? 
Thanks a lot
M.V.

**Reply by jhkennedy** [reply]
@MocioVileda the 688 is just the number of characters written to the file with `Path.write_text()`

As for the MintPy errors, it looks like the subsetted products weren't suitable for analysis with MintPy. Can you share the reference scene and SBAS parameters you used to generate your SBAS stack and submission to HyP3?

**Reply by MocioVileda** [reply]
Thank you @jhkennedy for your reply and help!
![image](https://user-images.githubusercontent.com/129189498/229042435-1411f3c3-a9ec-4b77-ac06-3a49a68f0fb3.png)
The polygon in yellow is my AOI, so I selected S1 track 168 descending to cover it. (SBAS ref. image e.g. S1A_IW_SLC__1SDV_20220601T052724_20220601T052751_043465_0530A1_148F)
I created the directory for my project folder as follows:

project_name = 'test_mintpy'
work_dir = Path.cwd() / project_name
data_dir = work_dir / 'data'
stack_start = parse_date('2022-06-01 00:00:00Z')
stack_end = parse_date('2022-09-30 00:00:00Z')
max_temporal_baseline = 48 #days
data_dir.mkdir(parents=True, exist_ok=True)

and set the import from ASF:

import asf_search as asf
import pandas as pd
search_results = asf.geo_search(
        platform=asf.SENTINEL1,
        intersectsWith='POLYGON((12.227 47.0821,12.186 47.0921,12.1189 47.0732,12.0968 47.0776,12.0767 47.0589,12.0399 47.0604,12.0213 47.0473,11.9791 47.05,11.9472 47.0335,11.9167 47.0332,11.8362 46.993,11.7818 46.9925,11.7656 46.9735,11.747 46.9689,11.7277 46.9724,11.7115 4,[...] ,12.1373 46.9795,12.1213 47.007,12.1474 47.0249,12.1644 47.0189,12.2049 47.027,12.2165 47.0587,12.2407 47.0692,12.227 47.0821))',
        start='2022-06-01',
        end='2022-09-30',
        processingLevel=asf.SLC,
        beamMode=asf.IW,
        flightDirection=asf.DESCENDING,
        relativeOrbit='168',
    )

The download succeded and I got the unwrap phase and coherence maps,
![coherence](https://user-images.githubusercontent.com/129189498/229043277-489efa4c-ba68-4768-815f-1bdef2cc3490.png)
![unwrapPhase](https://user-images.githubusercontent.com/129189498/229043288-89d239c9-79bc-4c11-8a30-e609ca80becf.png)
but I do not get the coherence matrix, coherence history, Temporal coherence...
![image](https://user-images.githubusercontent.com/129189498/229043713-d9fd3023-2a81-42c8-9aaa-6a7b58cbf48e.png)

Thank you for your help!!! 
M.V.

**Reply by jhkennedy** [reply]
@MocioVileda since your AOI is right on the edges of two scenes, I suspect you're having the same issues I outlined here:
https://github.com/insarlab/MintPy/discussions/985#discussioncomment-5394093

You'll need to be more selective when you subset the products.

**Reply by MocioVileda** [reply]
Thank you a lot for your suggestion @jhkennedy . I will try to refine the search and let you know. 
Thanks again!

M.V.

**Reply by MocioVileda** [reply]
I'm sorry to bother you again, but also refining the area selection I still get an error, a different one now: 
C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\objects\stack.py:917: RuntimeWarning: Mean of empty slice
  dmean[i] = np.nanmean(data)
C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\utils\plot.py:532: RuntimeWarning: All-NaN slice encountered
  ax.bar(x_list, np.nanmax(coh_mat, axis=0), bar_width.days, label='Max {}'.format(p_dict['ds_name']))
C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\utils\plot.py:533: RuntimeWarning: All-NaN slice encountered
  ax.bar(x_list, np.nanmin(coh_mat, axis=0), bar_width.days, label='Min {}'.format(p_dict['ds_name']))
C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\utils\plot.py:944: RuntimeWarning: All-NaN slice encountered
  vmax = int(np.nanmax(data))
Traceback (most recent call last):
  File "C:\Users\username\AppData\Local\anaconda3\Scripts\smallbaselineApp.py-script.py", line 9, in <module>
    sys.exit(main())
  File "C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\cli\smallbaselineApp.py", line 208, in main
    run_smallbaselineApp(inps)
  File "C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\smallbaselineApp.py", line 1117, in run_smallbaselineApp
    app.run(steps=inps.runSteps)
  File "C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\smallbaselineApp.py", line 883, in run
    self.run_quick_overview(sname)
  File "C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\smallbaselineApp.py", line 330, in run_quick_overview
    mintpy.cli.unwrap_error_phase_closure.main(iargs)
  File "C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\cli\unwrap_error_phase_closure.py", line 153, in main
    run_unwrap_error_phase_closure(inps)
  File "C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\unwrap_error_phase_closure.py", line 426, in run_unwrap_error_phase_closure
    out_file = calc_num_triplet_with_nonzero_integer_ambiguity(
  File "C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\unwrap_error_phase_closure.py", line 186, in calc_num_triplet_with_nonzero_integer_ambiguity
    pp.plot_num_triplet_with_nonzero_integer_ambiguity(out_file)
  File "C:\Users\username\AppData\Local\anaconda3\lib\site-packages\mintpy\utils\plot.py", line 944, in plot_num_triplet_with_nonzero_integer_ambiguity
    vmax = int(np.nanmax(data))
ValueError: cannot convert float NaN to integer
